src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```diff
@@ -375,12 +371,19 @@ private static string[] GetRootStoreDirectories(out bool isDefault)
         }
 
         private static bool TryStatFile(string path, out DateTime lastModified)
-            => TryStat(path, Interop.Sys.FileTypes.S_IFREG, out lastModified);
+            => CallStat(path, Interop.Sys.FileTypes.S_IFREG, out lastModified).HasCallSucceed;
 
         private static bool TryStatDirectory(string path, out DateTime lastModified)
-            => TryStat(path, Interop.Sys.FileTypes.S_IFDIR, out lastModified);
+            => CallStat(path, Interop.Sys.FileTypes.S_IFDIR, out lastModified).HasCallSucceed;
+
+        private static bool TryStatFile(string path, out DateTime lastModified, out (long Ino, long Dev) fileDetails)
```

Comment:
Let's avoid adding new functions, and keep using the _return bool for success_ pattern. The caller can ignore the `out` arguments (using `out _`) he doesn't need.
We'd have:
```cs
private static bool TryStatFile(string path, out DateTime lastModified, out (long, long) fileId)
   => CallStat(...
private static bool TryStatFile(string path, out DateTime lastModified)
   => CallStat(...
private static bool CallStat(string path, out DateTime lastModified, out (long, long) fileId)
```

Reply:
working on it in the next commit.. thank you for providing feedback